### ISSUE TITLE
[docs] How to write a custom task

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-tasks-core#151]
Add a docs page listing all requirements for using custom tasks.

Here below are the first ones that come to mind. I'm being very verbose, but a few of these requirements may be either relaxed or automatically checked.

1. Task interface:
    * The task should be associated to a python file which can be executed with the standard interface to be defined in https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/133 (to be placed in the `if __name__ == "__main__"` block).
    * Probably it is better that the task is self-contained, namely it does not depend on other python files which are not installed. But perhaps this constraint can be relaxed - to be checked. EDIT: if it is part of a package, it cannot use relative imports (like `from .mylib import myfunction`).
    * Note that the task command could be in principle an arbitrary executable, provided it has the same interface: instead of `/some/python task.py -j args.json -o out.json` we could have `/some/executable.x -j args.json -o out.json` (to be tested).
2. The first arguments of the function that is called in the `if __name__ == "__main__"` block must conform to a standard set (that will be something like `input_paths`, `output_paths`, and `metadata`, plus the optional `component`.
3. Output metadata:
    * If the task is to be executed sequentially, then the main function must return a dictionary (empty or not), which will be used to update the metadata during workflow execution.
    * The output dictionary of parallel tasks is not used in fractal, so don't put anything special in there.
    * The `history` key of the metadata dictionary is filled automatically by fractal, so that you should not modify it as part of your output.
4. The task should be listed in a manifest, conforming to fractal specs. EDIT: these are now being specified in https://github.com/fractal-analytics-platform/fractal-server/pull/188, and they come with a manifest version (not expected to evolve a lot, but it needs to be there to be future-proof).

EDIT: by now, they have to be pip installable.

**Comment by jluethi** [comment]
After the task refactor in https://github.com/fractal-analytics-platform/fractal-tasks-core/issues/300, we should update the how to part of writing a custom task to the docs.

I'll put any notes I observe here & questions of things I didn't find out, so we have a central place to collect them.

First question I had was how the args json file is structured that we're passing to a task when using the `run_fractal_task` wrapper function to run the task. I figured out that for the 0.8.0 task branch, they would look something like this I think for the create_ome_zarr task (so similar, but not the same as the args file we pass when adding a task to Fractal, now also including `input_paths`, `output_path` & `metadata`):

```json

{
    "input_paths": [
        "/Users/joel/Dropbox/Joel/FMI/Code/local_task_dev/fractal-tasks-core/examples/images/10.5281_zenodo.7057076"
    ],
    "output_path": "/Users/joel/Dropbox/Joel/FMI/Code/local_task_dev/fractal-tasks-core/examples/07_use_task_from_command_line/output/",
    "metadata": {},
    "num_levels": 5,
    "coarsening_xy": 2,
    "image_extension": "png",
    "metadata_table": "mrf_mlf",
    "allowed_channels": [
        {
            "colormap": "00FFFF",
            "wavelength_id": "A01_C01",
            "end": 800,
            "label": "DAPI",
            "start": 110
        },
        {
            "colormap": "FF00FF",
            "end": 290,
            "wavelength_id": "A01_C02",
            "label": "nanog",
            "start": 110
        },
        {
            "colormap": "FFFF00",
            "end": 1600,
            "wavelength_id": "A02_C03",
            "label": "Lamin B1",
            "start": 110
        }
    ]
}
```

**Comment by tcompa** [comment]
Note:
Depending on the packaging tool that one uses, it may be necessary to add specific options to include the `__FRACTAL_MANIFEST__.json` file in the package.

This can be checked explicitly, after installing the package in a test environment, e.g. as in
```
$ python -c 'import fractal_tasks_core; print(fractal_tasks_core.__file__)'
/somewhere/fractal_tasks_core/__init__.py

$ ls /somewhere/fractal_tasks_core/__FRACTAL_MANIFEST__.json 
/somewhere/fractal_tasks_core/__FRACTAL_MANIFEST__.json
```

**Comment by jluethi** [comment]
Thanks @tcompa 

Following up with some scmultiplex experiences: The `__FRACTAL_MANIFEST__.json` needed to be in the base folder of the source code (i.e. scmultiplex/src/scmultiplex/__FRACTAL_MANIFEST__.json), not in subfolders (I started with scmultiplex/src/scmultiplex/fractal/__FRACTAL_MANIFEST__.json).

To include it in the local wheel build, I then needed the following in the setup.cfg:
```
[options.package_data]
scmultiplex = __FRACTAL_MANIFEST__.json
```

Then I ran `python -m build` and `fractal task collect /path/to/scmultiplex-version-details.whl --package-extras fractal`

**Comment by tcompa** [comment]
The first version of this docs page is now ready (ref #482, which will be merged as part of #479)

It turned out to be quite a complex document, and reviews are welcome (both about clarity and contents).
@jluethi @gusqgm @ychiucco 

The file is currently visible at https://github.com/fractal-analytics-platform/fractal-tasks-core/blob/474-review-model-attributes-docstring-style/docs/custom_task.md, and will be part of `main` as soon as we merge #479.
